cloud/linode/cilium_loadbalancers.go

```diff
@@ -87,6 +88,22 @@ func (l *loadbalancers) getExistingSharedIPs(ctx context.Context) ([]string, err
 	return addrs, nil
 }
 
+func (l *loadbalancers) getExistingSharedIPsOnHolder(ctx context.Context) ([]string, error) {
+	addrs := []string{}
```

Comment:
This should be moved closer to where it's used, down to line 96/97. It can also be pre-allocated since you are copying `ipHolderAddrs.IPv4.Shared` into it.

Reply:
```suggestion
```